src/renderer/helpers/strings.js

```diff
@@ -56,6 +56,46 @@ export function translateWindowTitle(title) {
   }
 }
 
+/**
+ * @param {string} route
+ */
+export function getIconForRoute(route) {
+  const routeSlashIndex = route.indexOf('/', 2)
+  const truncatedRoute = (routeSlashIndex === -1) ? route : route.substring(0, routeSlashIndex)
+  switch (truncatedRoute) {
+    case '/subscriptions':
+      return ['fas', 'rss']
+    case '/subscribedchannels':
+    case '/channel':
+      return ['fas', 'list']
+    case '/trending':
+      return ['fas', 'fire']
+    case '/popular':
+      return ['fas', 'users']
+    case '/userplaylists':
+      return ['fas', 'bookmark']
+    case '/history':
+      return ['fas', 'history']
+    case '/settings':
+      return ['fas', 'sliders-h']
+    case '/about':
+      return ['fas', 'info-circle']
+    case '/search':
+      return ['fas', 'magnifying-glass']
+    case '/hashtag':
+      return ['fas', 'hashtag']
+    case '/post':
+      return ['fas', 'message']
+    case '/playlist': {
+      const solidOrRegular = route.includes('?playlistType=user') ? 'fas' : 'far'
+      return [solidOrRegular, 'bookmark']
+    } case '/watch':
+      return ['fas', 'play']
+    default:
+      return null
+  }
```

Comment:
I haven't tested this pull request yet but this list includes a lot of routes that there is no point bookmarking, as they have icons in the side bar.

I haven't looked into the latest changes on this pull request yet, but last time I looked at it I was planning to veto it entirely, as it was so far away from what it should have been, just saving search results like people asked. I had stopped myself just before posting the message as you were already demotivated enough and I didn't want you abandoning FreeTube, again.

Reply:
I have just seen your justifications above about you wanting to follow web browser behaviour. FreeTube is a desktop app, not a web browser, it makes sense to be able to bookmark any page in a web browser as you can open anything on the internet, there is no point in doing that in FreeTube and should frankly be considered broken behaviour.

Reply:
The reason I kept icons for all routes is because I intend to use these for navigation history. [See comment here.](https://github.com/FreeTubeApp/FreeTube/pull/5227#issuecomment-2515391099)

Reply:
> [I]t makes sense to be able to bookmark any page in a web browser [...] there is no point in doing that in FreeTube

This is not the current behavior in this PR, and this indicates to me that you have not actually tested it in its current state before reaching your conclusions. From what I can gather, from a purely functional perspective, your opinion is that the array below should just include `Search Results` and `Hashtag`, at which point some other tweaks to the implementation & UX could be made as well if we're 100% certain are the only two we'll ever be concerned about.
https://github.com/FreeTubeApp/FreeTube/blob/5ddfbcc1c439a1229bdb3313f8ef3ceb9d73cab4/src/renderer/components/top-nav/top-nav.js#L50-L58

My disagreement, that is seemingly shared by at least a few other team members, is that it doesn't present a detriment to the user experience to also allow users to save other relevant content like posts, channels, videos, playlists, and linkable settings sections / settings searches (TBD) for easy use if they so prefer. I'll provide counterarguments to three concerns to this approach now.

1. **Some people will not use this feature.** This is, in absence of additional harms applied to those users in exchange (a separate concern I will address later on), a neutral and acceptable state of affairs.
2. **Adding a new control will be confusing or otherwise materially harm the user experience by merit of it being present.** This is a valid concern insofar as the existence of any new control is something that can risk contributing to clutter or comprehension problems. This is where I believe therso far away from what it should have been, just saving search resulte was a misunderstanding with regards to my rebuttal I provided to this a while back:
> I have just seen your justifications above about you wanting to follow web browser behaviour [...] there is no point in doing that in FreeTube and should frankly be considered broken behaviour.

The claim I was making was not "if web browsers do it, let's copy what they do." Rather, I was addressing the comprehension concern that users would have comprehension issues learning what this feature is and how to use it, such that it would negatively contribute to their user experience. To clarify, this means that this is very easy to learn how to use and interact with as a control compared to any alternative access method that has been suggested so far. 

To bring it back to:
> It's so far away from what it should have been, just saving search results

What does the mental model look like for "save an arbitrary search so I can go back to it later?" and the corresponding subproblems? The supposedly more modest alternative is actually more foreign and harder to learn & use to most users! Whereas this is something that anyone knows how to use because we're appropriating the same mental model.

3. (**Philosophical claim) There is not a strict need for this, either because it has low salience or that there are / will be alternative access methods for these five additional types of content such that presenting this new access method is problematic.** Please correct me if I'm wrong, but I believe this is the main thing that you're hinting at with this quote:
> [T]here is no point in doing that in FreeTube and should frankly be considered broken behaviour.

This is a philosophical disagreement that I don't think you will readily change your mind on, but I do think I can at least try to address some relevant factors here.
  i. I do believe there is a bit of a personal bias in that you do not intend to use this feature yourself. This in itself fine, as I am also biased as someone who wants to use this feature, so I don't think holding that bias is inherently a bad thing.
  ii. Related to i, you think that the user base at large would feel the same. This one neither of us can discuss definitively because you and I both have dead zones for how we use the app & our priorities versus the user base at large. From a democratic perspective, let's say that we cancel each other out so far, thus leaving the rest of the team to be the deciders of ii.
  iii. Related to i and ii but not exactly, I do believe that part of this disagreement is that if we have other access methods for these forms of content (e.g., subscribing to a channel, saving a video to a playlist), adding a new one for them is buggy, wasteful, and/or a faux pas. This is something I addressed earlier, but what I disagree with in this philosophical claim is that it originates from a place of "if the users just interacted with the app the way we told them to, they wouldn't need this additional trash." Even if everything is possible through a rigid access method, that doesn't make it the only way that users can or should go about doing it. I'll link to [my earlier reply](https://github.com/FreeTubeApp/FreeTube/pull/5003#issuecomment-2146772725) where I addressed this in more depth, but needless to say, I do think there is room for serious disagreement with this philosophical claim.

As it stands, from the inventory above, my assessment is that most of your disagreement is represented by philosophical disagreement in point number 3 and its subclaims i-iii. Please correct me if I'm wrong on that, as I'm sure there might be more to your perspective, but you have not elucidated it yet, so I am going off of the available information you have provided. If I am correct that it is mainly philosophical concerns, I do think from a democratic perspective we should leave it to the rest of the team to determine whether to merge this or not, with you and me only providing advisory opinions in terms of the philosophical disagreements.